internal/expr/functions/builtins.go

```diff
@@ -652,3 +659,62 @@ func (s *AvgAggregator) Eval(_ *environment.Environment) (types.Value, error) {
 func (s *AvgAggregator) String() string {
 	return s.Fn.String()
 }
+
+// Len represents the len() function.
+// It returns the length of string, array or document.
+// For other types len() returns NULL.
+type Len struct {
+	Expr expr.Expr
+}
+
+// Eval extracts the average value from the given document and returns it.
+func (s *Len) Eval(env *environment.Environment) (types.Value, error) {
+	val, err := s.Expr.Eval(env)
+	if err != nil {
+		return nil, err
+	}
+	var length int
+	switch v := val.V().(type) {
+	case string:
```

Comment:
We should always use the `v.Type()` method and compare with the types defines in the types package (see [here](https://github.com/genjidb/genji/blob/487d4b4eec9331c86b4ead7d30aa4eb2e8b1e7f1/types/types.go#L22-L41)). The internal representation of values can sometimes differ (strings stored as []byte for example).

Example:

```go
switch t := val.Type() {
  case types.TextValue:
    // ...
  case types.DocumentValue:
   // ...
}
```

Reply:
I'll keep an eye on this and find a way to make sure we never create non-string TEXT values